tt_eager/tt_dnn/op_library/eltwise_binary/eltwise_binary_op.cpp

```diff
@@ -156,12 +156,12 @@ void EltwiseBinary::validate_with_output_tensors(const std::vector<Tensor>& inpu
         const auto& out_tensor = output_tensors.at(0).value();
         TT_FATAL(out_tensor.get_legacy_shape() == output_shape_required.at(0), fmt::format("The input tensors need a shape of {}, however the output tensor is only {}", output_shape_required,  out_tensor.get_legacy_shape()));
     }
-    if (this->in_place) {
-        TT_FATAL(input_tensor_a.memory_config().memory_layout == this->output_mem_config.memory_layout);
-        TT_FATAL(input_tensor_a.memory_config().buffer_type == this->output_mem_config.buffer_type);
+    if (this->in_place && this->output_mem_config.has_value()) {
+        TT_FATAL(input_tensor_a.memory_config().memory_layout == this->output_mem_config.value().memory_layout);
```

Comment:
We probably should have gotten rid of in_place before this mem_config change since this simplifies further changes

Reply:
okay. 
I hope everyone is okay with removing `in_place` flag from binary ops and replacing `tt_lib.operations.primary.add / ttnn.experimental.operations.primary.add` ?
I'll remove the inline Tensor add use binary `add() `with `output_tensor` i.e `tt_lib.tensor.add / ttnn.experimental.tensor.add` . This will involve changes in below files, shall I go ahead with it ? 

1. models/demos/t3000/falcon40b/tt/falcon_decoder.py
2. models/demos/resnet/tt/metalResnetBlock50.py
3. models/demos/t3000/llama2_70b/tt/llama_decoder_galaxy.py
4. models/demos/t3000/llama2_70b/tt/llama_decoder_optimized.py
5. models/experimental/llama2_70b/tt/llama_decoder_galaxy.py
6. models/experimental/llama2_70b/tt/llama_decoder_optimized.py
7. models/demos/falcon7b/tests/unit_tests/test_falcon_matmuls_and_bmms_with_mixed_precision.py